### PR TITLE
runtime metrics: batch statsd flushes

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

File: ddtrace/internal/runtime/runtime_metrics.py [hunk 1]
```diff
@@ -60,31 +60,20 @@ class RuntimeWorker(_worker.PeriodicWorkerThread):
...
-    def __init__(self, statsd_client, flush_interval=None):
-        flush_interval = self.FLUSH_INTERVAL if flush_interval is None else flush_interval
+    def __init__(self, statsd_client, flush_interval=FLUSH_INTERVAL):
         super(RuntimeWorker, self).__init__(interval=flush_interval,
                                             name=self.__class__.__name__)
         self._statsd_client = statsd_client
         self._runtime_metrics = RuntimeMetrics()
 
-    def _write_metric(self, key, value):
-        log.debug('Writing metric {}:{}'.format(key, value))
```
Comment (brettlangdon) [inline ddtrace/internal/runtime/runtime_metrics.py]: we lost this in the refactor

File: ddtrace/internal/runtime/runtime_metrics.py [hunk 2]
```diff
@@ -60,31 +60,20 @@ class RuntimeWorker(_worker.PeriodicWorkerThread):
...
-        self._statsd_client.gauge(key, value)
-
     def flush(self):
-        if not self._statsd_client:
```
Comment (brettlangdon) [inline ddtrace/internal/runtime/runtime_metrics.py]: was this just unnecessary?

Comment (jd) [reply ddtrace/internal/runtime/runtime_metrics.py]: I hope not. I mean we're not going to check every 10 seconds if we've been init correctly, right? 😓 
The `Tracer` always pass a stats client, fortunately.

File: ddtrace/internal/runtime/runtime_metrics.py (hunk 1 above)
Comment (jd) [reply ddtrace/internal/runtime/runtime_metrics.py]: I was not sure it was really useful; i'll add it back.

File: ddtrace/internal/runtime/runtime_metrics.py (hunk 2 above)
Comment (brettlangdon) [reply ddtrace/internal/runtime/runtime_metrics.py]: 👍

File: ddtrace/internal/runtime/runtime_metrics.py (hunk 1 above)
Comment (brettlangdon) [reply ddtrace/internal/runtime/runtime_metrics.py]: debug metric, cant hurt. saves us from having to use wireshark to debug these.

**Pull request merged**